packages/app/src/app/pages/Sandbox/Editor/Header/SandboxName/SandboxName.tsx

```diff
@@ -19,136 +25,130 @@ import {
-export const SandboxName = inject('store', 'signals')(
-  hooksObserver(
-    ({
-      store: {
-        isLoggedIn,
-        editor: { currentSandbox },
-      },
-      signals: {
-        modalOpened,
-        workspace: { sandboxInfoUpdated, valueChanged },
-      },
-    }) => {
-      const [updatingName, setUpdatingName] = useState(false);
-      const [name, setName] = useState('');
+export const SandboxName: FunctionComponent = () => {
+  const {
+    actions: {
+      modalOpened,
+      workspace: { sandboxInfoUpdated, valueChanged },
+    },
+    state: {
+      editor: { currentSandbox },
+      isLoggedIn,
+    },
+  } = useOvermind();
+  const [updatingName, setUpdatingName] = useState(false);
+  const [name, setName] = useState('');
 
-      const sandboxName = getSandboxName(currentSandbox) || 'Untitled';
+  const sandboxName = getSandboxName(currentSandbox) || 'Untitled';
 
-      const updateSandboxInfo = () => {
-        sandboxInfoUpdated();
-        setUpdatingName(false);
-      };
+  const updateSandboxInfo = () => {
+    sandboxInfoUpdated();
+    setUpdatingName(false);
+  };
 
-      const submitNameChange = (e: React.ChangeEvent<HTMLFormElement>) => {
-        e.preventDefault();
-        updateSandboxInfo();
-        track('Change Sandbox Name From Header');
-      };
+  const submitNameChange = (e: ChangeEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    updateSandboxInfo();
+    track('Change Sandbox Name From Header');
+  };
 
-      const handleNameClick = () => {
-        setUpdatingName(true);
-        setName(sandboxName);
-      };
+  const handleNameClick = () => {
+    setUpdatingName(true);
+    setName(sandboxName);
+  };
 
-      const handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
-        if (e.keyCode === ESC) {
-          updateSandboxInfo();
-        }
-      };
+  const handleKeyUp = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (e.keyCode === ESC) {
+      updateSandboxInfo();
+    }
+  };
 
-      const handleBlur = () => {
-        updateSandboxInfo();
-      };
+  const handleBlur = () => {
+    updateSandboxInfo();
+  };
 
-      const handleInputUpdate = (e: React.ChangeEvent<HTMLInputElement>) => {
-        valueChanged({
-          field: 'title',
-          value: e.target.value,
-        });
+  const handleInputUpdate = (e: ChangeEvent<HTMLInputElement>) => {
+    valueChanged({
+      field: 'title',
+      value: e.target.value,
+    });
 
-        setName(e.target.value);
-      };
+    setName(e.target.value);
+  };
 
-      const value = name !== 'Untitled' && updatingName ? name : '';
+  const value = name !== 'Untitled' && updatingName ? name : '';
 
-      const folderName = currentSandbox.collection
-        ? basename(currentSandbox.collection.path) ||
-          (currentSandbox.team ? currentSandbox.team.name : 'My Sandboxes')
-        : 'My Sandboxes';
+  const folderName = currentSandbox.collection
+    ? basename(currentSandbox.collection.path) ||
```

Comment:
@christianalfoni The types says `collection` is of type `boolean` so it can't have a `path`.
https://github.com/codesandbox/codesandbox-client/blob/master/packages/common/src/types/index.ts#L265-L287

I guess I should update the type? 🤔

Reply:
Oh, yeah, the way it works is that you change it in `packages/common/types/index.ts` And then in `packages/common` you can run `yarn build:lib`. You  might have to restart TS server to see changes in main project.

Great if you log out the `collection` there and try to figure out what kind of value is actually there to get typing beyond `path`, if there is anymore